reposilite-plugins/javadoc-plugin/src/main/kotlin/com/reposilite/plugin/javadoc/DocExtractor.kt

```diff
@@ -0,0 +1,56 @@
+package com.reposilite.plugin.javadoc
+
+import com.reposilite.web.http.ErrorResponse
+import com.reposilite.web.http.errorResponse
+import io.javalin.http.HttpCode
+import panda.std.Result
+import panda.std.Unit
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.jar.JarFile
+
+internal class DocExtractor {
+
+    fun extractJavadoc(jarFilePath: Path, destination: Path): Result<Unit, ErrorResponse> {
+        val file = jarFilePath.toFile()
+        val dest = destination.toFile()
+
+        // Some checks, to make sure we're working with valid files/paths.
+        require(!file.isDirectory) { "JavaDoc jar file must not be a directory!" }
+        require(dest.isDirectory) { "Destination must be a directory!" }
+        check(file.name.contains("doc.jar")) { "Invalid javadoc.jar! Name must contain: \"doc.jar\"" }
+
+        JarFile(file).use { jarFile ->
+            // Making sure we have an index.html file
+            val entry = jarFile.getEntry("index.html")
+            check(!(entry == null || entry.isDirectory)) { "Invalid doc.jar!" }
+
+            val result = extractJavadoc(destination.toFile(), jarFile)
+            file.delete()
+            return result
```

Comment:
Might be simplified with

```kotlin
return extractJavadoc(destination.toFile(), jarFile).also { /* delete */ }
```

Also, the `file` represents javadoc jar file, I don't think it should be deleted 🤔